### DISCUSSION TITLE
Does "ensemble.compute_ensemble" differ from manually DFT calculations?

### DISCUSSION THREAD

**Discussion by biglinn** [SSCHAcode/python-sscha#348] [Q&A]
Hi, @mesonepigreco:
The second tutorial gives a manual submission in the front part. This allows user to calculate energies and forces from generated configurations.
However, it also gives a automatical submission and calculates by the function `ensemble.compute_ensemble`.

I don't know if two ways are the same.

Moreover, there is a method `relax = sscha.Relax.SSCHA` to automatize all the procedures. User can perform NVT simulation with `relax.relax` and perform NPT simulation with `relax.vc_relax`.
This Relax module seemly also call `compute_ensemble` to perform calculations. How does it control NVT or NPT ensembles?

**Comment by mesonepigreco** [comment]
The NVT/NPT are the same relaxation at each population. The difference is that ``vc_relax`` uses the information on the stress tensor at the end of the minimization to change the primitive cell to reach the isotropic stress at the given target pressure, while ``relax`` just starts a new population with the same cell as the one used before.  

The ensemble calculation is the same for both NVT and NPT, that is why it does not change. 
The manual submission of the ensemble could be useful if you are using a calculator not implemented (e.g., VASP on a cluster). In this case you cannot use ``vc_relax`` or ``relax`` to do everything automatically, but you need to compute the ensemble and call one single step of ``vc_relax``  by choosing a starting pop >= than the maximum population (1 iteration) and telling the code to use the already computed ensemble (do not recompute it) using the ``restart_from_ens=True`` option.
```
relax.max_pop = 1
relax.vc_relax(target_press = 0, static_bulk_modulus = 50, start_pop = 1, restart_from_ens = True)
```

**Reply by biglinn** [reply]
So, `vc_relax(restart_from_ens = True)` skips calling `compute_ensemble` which perform calculations.

And `relax` directly starts a new population, does it mean that not using `relax` or `vc_realx` actually perform NVT as the same as using `relax`?
